src/clj_commons/byte_streams/graph.clj

```diff
@@ -122,7 +122,7 @@
                    (assoc-in [(Type. 'seq src) (Type. 'seq dst)]
                      (Conversion. (fn [x options] (map #(f % options) x)) cost))
                    (assoc-in [(Type. 'stream src) (Type. 'stream dst)]
-                     (Conversion. (fn [x options] (s/map #(f % options) x)) (+ cost 0.1)))))
+                     (Conversion. (fn [x options] (s/map #(f % options) x)) (+ ^double cost 0.1)))))
```

Comment:
Seems like the hint should be higher up, but it's not working. I forget if this is a limitation of deftype or not.

Probably preferable to convert at the top of the let, like:

```clojure
(let [cost (double cost)
      m' (assoc-in ...
```

Reply:
Or a limitation of `defprotocol`, I mean, which defines IConversionGraph. (Which should be named something else, since it's not an interface.)

Reply:
Yeah I tried to add the hint both on defprotocol and deftype but it errored...

Reply:
Not surprised. I get the compromises necessary to interop with Java, but the whole deftype/protocol/record system is ugly and sticks out.